show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,259 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showlineofsightbetweengeoelements.components
+
+import android.app.Application
+import androidx.core.content.ContextCompat.getString
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.analysis.GeoElementLineOfSight
+import com.arcgismaps.geometry.AngularUnit
+import com.arcgismaps.geometry.GeodeticCurveType
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.LinearUnit
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.PointBuilder
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ArcGISTiledElevationSource
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Surface
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISSceneLayer
+import com.arcgismaps.mapping.symbology.ModelSceneSymbol
+import com.arcgismaps.mapping.symbology.SceneSymbolAnchorPosition
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.arcgismaps.mapping.view.AnalysisOverlay
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SurfacePlacement
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import com.esri.arcgismaps.sample.showlineofsightbetweengeoelements.R
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import java.util.Timer
+import java.util.TimerTask
+
+class SceneViewModel(private var application: Application) : AndroidViewModel(application) {
+
+
+    // Create a scene and add a basemap to it
+    val scene = ArcGISScene(BasemapStyle.ArcGISTopographic)
+
+    // Create graphic overlay to hold graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    // Create an analysis overlay to hold the line of sight
+    val analysisOverlay = AnalysisOverlay()
+
+    // Initialize z to 50 as starting point and observe any changes to its value
+    private val _currentZValue = MutableStateFlow(50.0)
+    val currentZValue: StateFlow<Double> = _currentZValue.asStateFlow()
+
+    // The following variables keep track of the taxi's location
+    private var waypointsIndex = 0
+    private val meters = LinearUnit.meters
+    private val degrees = AngularUnit.degrees
```

Comment:
These aren't necessarily used to hold a value as ViewModel functions are being called, so it probably shouldn't be a top-level `val`. Instead, you can use these geometry units directly when needed for an SDK call. 

```suggestion
```